File: src/main/java/com/qwwuyu/server/dao/NoteMapper.java
package com.qwwuyu.server.dao;

import com.qwwuyu.server.bean.Note;
import org.apache.ibatis.annotations.Param;

import java.util.List;

public interface NoteMapper {
    Note selectByPrimaryKey(Integer id);

    int insert(Note record);

    int updateByPrimaryKeySelective(Note record);

    int deleteByPrimaryKey(Integer id);

    List<Note> selectByNote(@Param("note") Note note, @Param("limit") int limit, @Param("offset") int offset, @Param("asc") String asc, @Param("desc") String desc);
}

File: app/src/main/kotlin/com/qwwuyu/gs/mapper/NoteMapper.kt
package com.qwwuyu.gs.mapper

import com.qwwuyu.gs.entity.Note
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface NoteMapper {
    fun selectByPrimaryKey(id: Int): Note?

    fun insert(record: Note): Int

    fun updateByPrimaryKeySelective(record: Note): Int

    fun deleteByPrimaryKey(id: Int): Int

    fun selectByNote(@Param("note") note: Note, @Param("limit") limit: Int, @Param("offset") offset: Int,
                     @Param("asc") asc: String?, @Param("desc") desc: String?): List<Note>
}